app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function Ga4EnhancedEcommerceTracker (module) {
+    this.module = module
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.init = function (isNewPage = true) {
+    var consentCookie = window.GOVUK.getConsentCookie()
+    this.isNewPage = isNewPage
+
+    if (consentCookie && consentCookie.settings) {
+      this.startModule()
+    } else {
+      this.startModule = this.startModule.bind(this)
+      window.addEventListener('cookie-consent', this.startModule)
+    }
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.startModule = function () {
+    this.PIIRemover = new window.GOVUK.analyticsGA4.PIIRemover()
+    this.DEFAULT_LIST_TITLE = 'Site search results'
+
+    window.dataLayer = window.dataLayer || []
+
+    if (this.isNewPage) {
+      this.module.addEventListener('click', this.handleClick.bind(this))
+    } else {
+      if (window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker) {
+        var pageViewTracker = window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker
+        window.dataLayer.push({
+          event: 'page_view',
+          page_view: {
+            location: pageViewTracker.getLocation(),
+            referrer: pageViewTracker.getReferrer()
```

Comment:
I'm a bit confused by this section of code - it looks like you're getting the page view code and using part of it to create a new page view in here? I feel like this could increase maintenance if we decide to change the data structure of page views. Instead, could you call the page view code directly?

```JavaScript
var pageViewTracker = window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker
if (pageViewTracker) {
  pageViewTracker.init()
}
```

Reply:
As per our discussion earlier, I've implemented your above suggestion with a view to refactor the `PageViewTracker` code at a later date to accommodate scenarios where we only need to push a few select values of the `pageView` object to the `dataLayer` rather than the whole thing.